misk-metrics/src/main/kotlin/misk/metrics/v3/Metrics.kt

```diff
@@ -0,0 +1,257 @@
+   *  for the metric. The key of the map is the quantile as a ratio (e.g. 0.99 represents p99) and
+   *  the value is the "tolerable error" of the computed quantile.
+   */
+  @JvmOverloads
+  fun summary(
+    name: String,
+    description: String = "",
+    tags: Iterable<Tag> = listOf(),
+    quantiles: Map<Double, Double> = defaultQuantiles,
+    maxAgeSeconds: Long = TimeUnit.MINUTES.toMillis(10)
+  ): DistributionSummary {
+
+    val precision = quantiles.values.maxOfOrNull { it.toString().substringAfter('.').length } ?: 4
+    return DistributionSummary.builder(name)
+      .description(description)
+      .tags(tags)
+      .publishPercentiles(*quantiles.keys.toDoubleArray())
+
+      .percentilePrecision(precision)
+      // The default for this in prometheus was 10 minutes (above), default in micrometer is 2 minutes
+      .distributionStatisticExpiry(Duration.ofMillis(maxAgeSeconds))
+      // this is the default in prometheus. The default in micrometer is 3
+      .distributionStatisticBufferLength(5)
+      .register(registry)
+  }
+
+  companion object {
+
+    /**
+     * A peak gauge is a wrapper around an AtomicDouble that resets to an initial value of 0 after
+     * observation (i.e. after a metric collection.)
+     *
+     * This is useful for accurately capturing maximum observed values over time. In contrast to the
+     * histogram maximum which tracks the maximum value in its sampling window. That sampling window
+     * typically covers multiple metric collections.
+     */
+    class PeakGauge {
+      // For simplicity, using an atomic double with an initial value of 0 here.
+      //
+      // If we cared to differentiate between the initial value due to no samples vs having
+      // samples that are equal or less than 0, then we would need to also track whether we
+      // have received an update since the last reset.
+      private val value = AtomicDouble()
+
+      /**
+       * Updates the stored value if the new value is greater.
+       */
+      fun record(newValue: Double) {
+        while (true) {
+          val previous = value.get()
+          if (newValue > previous) {
+            value.compareAndSet(previous, newValue)
+          } else {
+            return
+          }
+        }
+      }
+
+      /**
+       * Reset to the initial value and return previously held value.
+       */
+      fun getAndClear(): Double {
+        return value.getAndSet(0.0)
+      }
+    }
+  }
+}
+
+val defaultQuantiles = mapOf(
+  0.5 to 0.05,
+  0.75 to 0.02,
+  0.95 to 0.01,
+  0.99 to 0.001,
+  0.999 to 0.0001
+)
+
+/**
+ * defaultBuckets is a default set of buckets which assumes the value is in milliseconds (ms).
+ *
+ * It contains 58 buckets which range from 1ms to 1hr.
+ *
+ * Adapted from the default M3 buckets.
+ *
+ * https://github.com/m3db/m3/blob/v1.1.0/src/x/instrument/methods.go#L57-L83
+ */
+val defaultBuckets = listOf(
```

Comment:
Nit: Could this be called `millisBuckets` then, or some such?

Reply:
It's default buckets because that's what exists in v2. We actually don't want buckets in micrometer, it's an anti-pattern. The bucketing I put into the micrometer versions of things is to keep things consistent until we can do a separate migration away from that, as I'm concerned that it may change things in an unexpected way.

Reply:
Could it not still be called `millisBuckets` though? We had a discussion about this for the v2 API about 6 months ago. @ericloe-cash was working on a metrics helper to improve bucketing in misk metrics, not sure where that ended up though.